packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -844,59 +844,77 @@ export default {
-            `${reactiveHookName} Hook (at line ${declaredDependenciesNode.loc.start.line}) ` +
-            `change on every render.`;
-          if (suggestUseCallback) {
-            message +=
-              ` To fix this, ` +
-              `wrap the '${fn.name.name}' definition into its own useCallback() Hook.`;
-          } else {
-            message +=
-              ` Move it inside the ${reactiveHookName} callback. ` +
-              `Alternatively, wrap the '${fn.name.name}' definition into its own useCallback() Hook.`;
-          }
-
-          let suggest;
-          // Only handle the simple case: arrow functions.
-          // Wrapping function declarations can mess up hoisting.
-          if (suggestUseCallback && fn.type === 'Variable') {
-            suggest = [
-              {
-                desc: `Wrap the '${fn.name.name}' definition into its own useCallback() Hook.`,
-                fix(fixer) {
-                  return [
-                    // TODO: also add an import?
-                    fixer.insertTextBefore(fn.node.init, 'useCallback('),
-                    // TODO: ideally we'd gather deps here but it would require
-                    // restructuring the rule code. This will cause a new lint
-                    // error to appear immediately for useCallback. Note we're
-                    // not adding [] because would that changes semantics.
-                    fixer.insertTextAfter(fn.node.init, ')'),
-                  ];
+        constructions.forEach(
+          ({construction, isUsedOutsideOfHook, depType}) => {
+            const wrapperHook =
+              depType === 'function' ? 'useCallback' : 'useMemo';
+
+            const defaultAdvice = `wrap the construction of '${construction.name.name}' in its own ${wrapperHook}() Hook.`;
+
+            const advice = isUsedOutsideOfHook
+              ? `To fix this, ${defaultAdvice}`
+              : `Move it inside the ${reactiveHookName} callback. Alternatively, ${defaultAdvice}`;
+
+            const causation =
+              depType === 'conditional' || depType === 'logical expression'
+                ? 'could make'
+                : 'makes';
+
+            const message =
+              `The '${construction.name.name}' ${depType} ${causation} the dependencies of ` +
+              `${reactiveHookName} Hook (at line ${declaredDependenciesNode.loc.start.line}) ` +
+              `change on every render. ${advice}`;
+
+            let suggest;
+            // Only handle the simple case of variable assignments.
+            // Wrapping function declarations can mess up hoisting.
+            if (
+              isUsedOutsideOfHook &&
+              construction.type === 'Variable' &&
+              // Objects may be mutated ater construction, which would make this
+              // fix unsafe. Functions _probably_ won't be mutated, so we'll
+              // allow this fix for them.
+              depType === 'function'
```

Comment:
won't this only turn on the suggestion for functions?

Reply:
I hoped that the comment above would explain the reasoning.

Reply:
oops. I misread it. all good!